crates/optimism/rpc/src/supervisor.rs

```diff
@@ -0,0 +1,59 @@
+//! This is our custom implementation of validator struct
+
+use jsonrpsee::http_client::{HttpClient, HttpClientBuilder};
+use kona_interop::{ExecutingDescriptor, SafetyLevel};
+use kona_rpc::InteropTxValidator;
+use std::{sync::Arc, time::Duration};
+/// Implementation of the supervisor trait for the interop.
+#[derive(Debug, Clone)]
+pub struct SupervisorClient {
+    inner: Arc<SupervisorClientInner>,
+}
+
+impl SupervisorClient {
+    /// Creates a new supervisor validator.
+    pub fn new(supervisor_endpoint: impl Into<String>) -> Self {
+        let url = Url::parse(supervisor_endpoint.into()).expect("parsing supervisor url");
+        let client =
+            HttpClientBuilder::default().build(url).expect("building supervisor http client");
+        Self::with_client(supervisor_endpoint, client)
+    }
+
+    /// Creates a new supervisor validator.
+    pub fn with_client(supervisor_endpoint: impl Into<String>, client: HttpClient) -> Self {
+        let inner =
+            SupervisorClientInner { supervisor_endpoint: supervisor_endpoint.into(), client };
+        Self { inner: Arc::new(inner) }
+    }
+
+    pub fn parse_access_list(access_list_items: &[AccessListItem]) -> impl Iterator<Item = &B256> {
+        self.inner.parse_access_list(access_list_items)
+    }
+
+    /// Validates a list of inbox entries against a Supervisor.
+    pub async fn validate_messages(
+        &self,
+        inbox_entries: &[B256],
+        safety: SafetyLevel,
+        executing_descriptor: ExecutingDescriptor,
+        timeout: Option<Duration>,
+    ) -> Result<(), InteropTxValidatorError> {
+        self.inner.validate_messages(inbox_entries, safety, executing_descriptor, timeout).await
+    }
+}
+
+pub struct SupervisorClientInner {
+    /// The endpoint of the supervisor
+    supervisor_endpoint: String,
```

Comment:
this is only needed once to instantiate the client, right? no need to store it here

I general I think this inner type is redundant and we can just do 
```rust
pub struct SupervisorClient<T> {
    inner: T
}
```
we want the generic type `T` so we can also use web socket protocol. then we do blanket impl
```rust
impl<T> InteropTxValidator for SupervisorClient<T> where T: SupervisorApiClient {
   type SupervisorClient = T;
   ..
}
```

Reply:
But I also probably will add some king or reconnection in here
Aka got connection error - construct new inner

Reply:
add that as separate pr later, if you manage to get to that point - prios are constantly changing